File: frontend/user/lib/data/mapper/order/delivery_info_mapper.dart
import 'package:b2205946_duonghuuluan_luanvan/data/models/order/delivery_info_model.dart';
import 'package:b2205946_duonghuuluan_luanvan/domain/entity/order/delivery_info.dart';

class DeliveryInfoMapper {
  static DeliveryInfo fromModel(DeliveryInfoModel model) {
    return DeliveryInfo(
      id: model.id,
      userId: model.userId,
      name: model.name,
      address: model.address,
      phone: model.phone,
      districtId: model.districtId,
      wardCode: model.wardCode,
    );
  }
}

File: backend/app/infrastructure/database/mappers/delivery_info_mapper.py
from app.domain.entities.delivery_info_entity import DeliveryInfoEntity


class DeliveryInfoMapper:
    @staticmethod
    def to_entity(model) -> DeliveryInfoEntity:
        return DeliveryInfoEntity(
            id=model.id,
            user_id=model.user_id,
            name=model.name,
            address=model.address,
            phone=model.phone,
            district_id=model.district_id,
            ward_code=model.ward_code,
            default=model.default,
            created_at=model.created_at,
            updated_at=model.updated_at,
        )
